sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -1757,7 +1757,7 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
       sql("insert into tbl values ('3', '2.3')")
       checkAnswer(
         sql("select (cast (99 as decimal(19,6)) + cast('3' as decimal)) * cast('2.3' as decimal)"),
-        Row(204.0)
+        Row(BigDecimal(204.0))
```

Comment:
the result is decimal type, not double.

Reply:
It's good to fix this.
Do you think this PR also affects the Spark's behavior on the existing apps?

Reply:
It does not.

This test case was wrong at the first place, my change to the `checkAnswer` expose it.